src/IO/ReadWriteBufferFromHTTP.h

```diff
@@ -743,19 +790,46 @@ namespace detail
         const std::string & getCompressionMethod() const { return content_encoding; }
 
         std::optional<time_t> getLastModificationTime()
+        {
+            return getFileInfo().last_modified;
+        }
+
+        FileInfo getFileInfo()
         {
             Poco::Net::HTTPResponse response;
             getHeadResponse(response);
-            if (!response.has("Last-Modified"))
-                return std::nullopt;
+            return parseFileInfo(response, 0);
+        }
+
+        FileInfo parseFileInfo(const Poco::Net::HTTPResponse & response, size_t requested_range_begin)
+        {
+            FileInfo res;
 
-            String date_str = response.get("Last-Modified");
-            struct tm info;
-            char * res = strptime(date_str.data(), "%a, %d %b %Y %H:%M:%S %Z", &info);
-            if (!res || res != date_str.data() + date_str.size())
-                return std::nullopt;
+            if (response.hasContentLength())
+            {
+                res.file_size = response.getContentLength();
 
-            return timegm(&info);
+                if (response.getStatus() == Poco::Net::HTTPResponse::HTTPStatus::HTTP_PARTIAL_CONTENT)
+                {
+                    *res.file_size += requested_range_begin;
+                    res.seekable = true;
+                }
+                else
+                {
+                    res.seekable = response.has("Accept-Ranges") && response.get("Accept-Ranges") == "bytes";
+                }
```

Comment:
This changes behavior slightly: previously StorageURL would send a HEAD request with "Range: 0-" for this check, now it's without "Range" (except if this parseFileInfo() is called from nextImpl(), which is not the important call site). I expect it's ok because HTTP servers are supposed to report "Accept-Ranges: bytes" anyway (?), but I don't actually know what HTTP servers are out there and what headers they send. For S3 this works, for althttpd this didn't work even before the change, didn't try anything else.